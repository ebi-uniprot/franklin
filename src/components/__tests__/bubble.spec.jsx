import { render } from '@testing-library/react';

import Bubble from '../bubble';

describe('Long number component', () => {
  test('should render with defaults', () => {
    const { asFragment } = render(<Bubble value={10} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render bigger than 100', () => {
    const { asFragment } = render(<Bubble value={101} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render different size and colour', () => {
    const { asFragment } = render(
      <Bubble value={101} colourClass="colour-light-grey" size="large" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
