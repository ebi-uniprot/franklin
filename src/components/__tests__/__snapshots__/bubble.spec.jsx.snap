// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Long number component should render bigger than 100 1`] = `
<DocumentFragment>
  <span
    class="bubble--medium colour-sea-blue"
  >
    99+
  </span>
</DocumentFragment>
`;

exports[`Long number component should render different size and colour 1`] = `
<DocumentFragment>
  <span
    class="bubble--large colour-light-grey"
  >
    99+
  </span>
</DocumentFragment>
`;

exports[`Long number component should render with defaults 1`] = `
<DocumentFragment>
  <span
    class="bubble--medium colour-sea-blue"
  >
    10
  </span>
</DocumentFragment>
`;
